.github/workflows/go.yml

```diff
@@ -308,7 +308,9 @@ jobs:
         run: go install honnef.co/go/tools/cmd/staticcheck@${{ matrix.staticcheck }}
       - name: Build
         shell: bash
-        run: ci/scripts/go_build.sh $(pwd)
+        run: |
+           echo "PKG_CONFIG_PATH=/usr/local/Cellar/openssl@3/3.0.8/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
```

Comment:
This doesn't affect in this step.

We need a separated step:

```yaml
- name: Prepare PKG_CONFIG_PATH
  shell: bash
  run: |
    echo "PKG_CONFIG_PATH=$(brew --prefix openssl)/lib/pkgconfig:$PKG_CONFIG_PATH" >> $GITHUB_ENV
- name: Build
  shell: bash
  run: ci/scripts/go_build.sh $(pwd)
...
```

And this change causes YAML syntax error: https://github.com/apache/arrow/actions/runs/4358242702/workflow#L308

I think that the indentation of this line is a problem.

Reply:
Also weird how the workflow didn't run on the last commit. Sorry for the indentation error that happens due to my comment suggestion 🙈